### PR TITLE
fix(data-management): Fix event definition ordering

### PR DESCRIPTION
**Pull request by timgl**
## Problem

Event definition only orders per page, not across the whole list of event definitions

## Changes

order in backend

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by pauldambra** [comment]
~~Weirdly this is the same error that I hit before I was distracted by other work... 30 day query count doesn't sort as expected...~~

🤦 It's not a bug... we set `NULLS LAST` so if everything except a small number of values is null then it looks odd (to me)

File: frontend/src/scenes/data-management/events/eventDefinitionsTableLogic.ts [hunk 1]
```diff
@@ -25,13 +25,15 @@ export interface Filters {
...
     event_type: EventDefinitionType
+    ordering?: string
 }
 
 function cleanFilters(filter: Partial<Filters>): Filters {
     return {
         event: '',
         properties: [],
         event_type: EventDefinitionType.Event,
+        ordering: '-volume_30_day',
```
Comment (pauldambra) [inline frontend/src/scenes/data-management/events/eventDefinitionsTableLogic.ts:36]: I think we should use `order` to stay similar to other parts of the API

File: posthog/api/utils.py [hunk 2]
```diff
@@ -364,11 +368,7 @@ def select_event_definitions(fields: str):
 
     # Only return event definitions
     raw_event_definition_fields = ",".join(event_definition_fields)
-    ordering = (
-        "ORDER BY last_seen_at DESC NULLS LAST, query_usage_30_day DESC NULLS LAST, name ASC"
-        if is_enterprise
-        else "ORDER BY name ASC"
-    )
+    ordering = f"ORDER BY {order_UNSAFE} {order_direction} NULLS LAST, name ASC"
```
Comment (pauldambra) [inline posthog/api/utils.py:371]: Am torn on whether we should force nulls last.

But, not blocking. Some server sorting is better than none

**Review by pauldambra — APPROVED** [review]
We should probably use `order` rather than `ordering` but 🤷

File: posthog/api/utils.py [hunk 3]
```diff
@@ -333,7 +356,11 @@ def safe_clickhouse_string(s: str) -> str:
 
 
 def create_event_definitions_sql(
-    event_type: EventDefinitionType, is_enterprise: bool = False, conditions: str = ""
+    event_type: EventDefinitionType,
+    is_enterprise: bool = False,
+    conditions: str = "",
+    order_UNSAFE: str = "",
+    order_direction: str = "DESC",
```
Comment (Twixes) [inline posthog/api/utils.py:363]: `order_UNSAFE` and `order_direction` is kind of confusing. I mean, "order" is more general than "order_direction", yet both are here at the same time… and both are equally unsafe. BTW the correct-er type for `order_direction` would be `Literal["DESC", "ASC"]`

File: posthog/api/utils.py [hunk 4]
```diff
@@ -364,11 +391,10 @@ def select_event_definitions(fields: str):
 
     # Only return event definitions
     raw_event_definition_fields = ",".join(event_definition_fields)
-    ordering = (
-        "ORDER BY last_seen_at DESC NULLS LAST, query_usage_30_day DESC NULLS LAST, name ASC"
-        if is_enterprise
-        else "ORDER BY name ASC"
+    provided_ordering = (
+        f"{order_UNSAFE} {order_direction} {'NULLS FIRST' if order_direction == 'ASC' else 'NULLS LAST'}"
```
Comment (Twixes) [inline posthog/api/utils.py:395]: `order_UNSAFE` can be empty (and is empty by default), in which case this will error out.

**Review by Twixes — COMMENTED** [review]
+1 for the fix, only `create_event_definitions_sql` got a bit messy

**Pull request merged**